tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -3216,6 +3216,13 @@ SpirvInstruction *SpirvEmitter::processBufferTextureLoad(
     return nullptr;
   }
 
+  if (residencyCode && residencyCode->isRValue()) {
```

Comment:
What happens if `residencyCode` is `nullptr`?
Why don't we change this condition to `if (residencyCode == nullptr || residencyCode->isRValue()) {`?

Reply:
this function (`SpirvEmitter::processBufferTextureLoad`) handles the Load methods. There are 2 Load methods that are defined for textures and buffer: one method that just does the load, and one method that does the load and also returns the status of the operation.

For example, for Texture2D we have these 2 methods: ([spec](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/texture2d-load))
```
Load(int,int) : Reads texture data.
Load(int,int,uint) : Reads texture data and returns status of the operation.
```
In the first method, `residencyCode` is in fact `nullptr` and it should not be an error.

Reply:
Got it. It would be better to add a comment to briefly explain it.